### ISSUE TITLE
cannot manipulate tag named '--'

### ISSUE THREAD

**Issue by 0ion9** [oniony/TMSU#63]
I don't actually want a tag named `--`. However, it is possible to accidentally create one via
`tmsu tag --tags -- "${SOMEFILES[@]}"`. At that point you cannot run `tmsu delete --` (or `tmsu delete -- --`, which IMO  would be correct -- first `--` is argument separator, second `--` is argument). Similar problems exist with `merge` and `rename` subcommands.

The simplest workaround is to use sqlite3 directly on the DB to adjust the name to something that can be deleted/merged, like `000` -- eg `update tag set name="000" where name="--";`

**Comment by oniony** [comment]
Thanks for reporting it. I think you're right with repercussions to first
-- ending parsing of options. It's a bit of an unlikely edge case but
nevertheless it shouldn't work as it currently does so I'll get a fix for
it.

**Comment by oniony** [comment]
This should now be fixed. I have not prevented the creation of this tag, i.e. if the parser is expecting an option for an argument, then it still does not treat `--` specially. However I have fixed the bug that meant that `--` was always swallowed, no matter how many times it was specified on the command-line.

I have also raised issue #65 for replacing the home-spun command-line parsing with a third-party library.
